src/node/genesis_gen.h

```diff
@@ -398,10 +398,41 @@ namespace ccf
       shares_view->put(0, key_share_info);
     }
 
-    void set_recovery_threshold(size_t threshold)
+    bool set_recovery_threshold(size_t threshold)
     {
       auto config_view = tx.get_view(tables.config);
+
+      // While waiting for recovery shares, the recovery threshold cannot be
+      // modified. Otherwise, the threshold could be passed without triggering
+      // the end of recovery procedure
+      if (
+        get_service_status().value() ==
+        ServiceStatus::WAITING_FOR_RECOVERY_SHARES)
+      {
+        LOG_FAIL_FMT(
+          "Failed to set recovery threshold: service is currently waiting for "
```

Comment:
I think "Cannot adjust recovery threshold while waiting for recovery shares" would make it clearer that is a restriction/validation error rather than something that didn't work.